File: Angular Client/models/visit.ts
import {User, UserEntity} from './user';
import {FormGroup} from '@angular/forms';
import {PreventiveTreatment} from './preventive-treatment';

export interface Visit {
  visitId: number;
  animalId: number;
  visitTime: Date;
  cause: string;
  symptoms: string;
  labResults: string;
  comment: string;
  temperature: number;
  weight: number;
  pulse: number;
  diagnosis1: string;
  diagnosis2: string;
  diagnosis3: string;
  treatment1: string;
  treatment2: string;
  treatment3: string;
  treatment4: string;
  treatment5: string;
  treatment6: string;
  createdDate: Date;
  archive: boolean;
  preventiveTreatmentsList: PreventiveTreatment[];
  userId: number;
  user: User;
}

export class VisitEntity implements Visit{
  constructor() {
    this.animalId = 0;
    this.archive =  false;
    this.cause =  '';
    this.comment =  '';
    this.createdDate = new Date();
    this.diagnosis1 =  '';
    this.diagnosis2 =  '';
    this.diagnosis3 =  '';
    this.labResults =  '';
    this.pulse = 0;
    this.symptoms =  '';
    this.temperature = 0;
    this.treatment1 =  '';
    this.treatment2 =  '';
    this.treatment3 =  '';
    this.treatment4 =  '';
    this.treatment5 =  '';
    this.treatment6 =  '';
    this.user = new UserEntity();
    this.userId = 0;
    this.visitId = 0;
    this.visitTime = new Date();
    this.weight = 0;
    this.preventiveTreatmentsList = [];
  }
  animalId: number;
  archive: boolean;
  cause: string;
  comment: string;
  createdDate: Date;
  diagnosis1: string;
  diagnosis2: string;
  diagnosis3: string;
  labResults: string;
  pulse: number;
  symptoms: string;
  temperature: number;
  treatment1: string;
  treatment2: string;
  treatment3: string;
  treatment4: string;
  treatment5: string;
  treatment6: string;
  user: User;
  userId: number;
  visitId: number;
  visitTime: Date;
  weight: number;
  preventiveTreatmentsList: PreventiveTreatment[];

  setData(form: FormGroup): void{
    this.animalId = form.get('animalId')?.value;
    this.cause = form.get('cause')?.value;
    this.comment = form.get('comment')?.value;
    this.diagnosis1 = form.get('diagnosis1')?.value;
    this.diagnosis2 = form.get('diagnosis2')?.value;
    this.diagnosis3 = form.get('diagnosis3')?.value;
    this.labResults = form.get('labResults')?.value;
    this.symptoms = form.get('symptoms')?.value;
    this.pulse = form.get('pulse')?.value;
    this.temperature = form.get('temperature')?.value;
    this.weight = form.get('weight')?.value;
    this.treatment1 = form.get('treatment1')?.value;
    this.treatment2 = form.get('treatment2')?.value;
    this.treatment3 = form.get('treatment3')?.value;
    this.treatment4 = form.get('treatment4')?.value;
    this.treatment5 = form.get('treatment5')?.value;
    this.treatment6 = form.get('treatment6')?.value;
    this.visitId = form.get('visitId')?.value;
    this.visitTime = new Date(form.get('visitTime')?.value);
  }
}

File: ASP.NET Core Server/Petadmin.Core/Models/Visit.cs
using System;
using System.Collections.Generic;
using PetAdmin.Core.Interfaces;

namespace Petadmin.Core.Models
{
    public class Visit : IGenericDbEntity
    {
        /// <summary>
        /// Keeps the database id of the visit.
        /// </summary>
        public int VisitId { get; set; }

        /// <summary>
        /// Keeps the database id of the animal.
        /// </summary>
        public int AnimalId { get; set; }

        /// <summary>
        /// Keeps the database id of the admin user.
        /// </summary>
        public int UserId { get; set; }

        /// <summary>
        /// Gets or sets the time of visit of the animal.
        /// </summary>
        public DateTime? VisitTime { get; set; }

        /// <summary>
        /// Gets or sets the date of next visit of the animal.
        /// </summary>
        public DateTime? NextVisitDate { get; set; }

        /// <summary>
        /// Gets or sets the cause of the visit.
        /// </summary>
        public string Cause { get; set; }

        /// <summary>
        /// Gets or sets the symptoms for the visit.
        /// </summary>
        public string Symptoms { get; set; }

        /// <summary>
        /// Gets or sets the laboratory results for the visit.
        /// </summary>
        public string LabResults { get; set; }

        /// <summary>
        /// Gets or sets the comment of the visit.
        /// </summary>
        public string Comment { get; set; }

        /// <summary>
        /// Gets or sets the temperature for the visit.
        /// </summary>
        public decimal Temperature { get; set; }

        /// <summary>
        /// Gets or sets the weight for the visit.
        /// </summary>
        public decimal Weight { get; set; }

        /// <summary>
        /// Gets or sets the pulse for the visit.
        /// </summary>
        public int Pulse { get; set; }

        /// <summary>
        /// Gets or sets the diagnosis1 of the visit.
        /// </summary>
        public string Diagnosis1 { get; set; }

        /// <summary>
        /// Gets or sets the diagnosis2 of the visit.
        /// </summary>
        public string Diagnosis2 { get; set; }

        /// <summary>
        /// Gets or sets the diagnosis3 of the visit.
        /// </summary>
        public string Diagnosis3 { get; set; }

        /// <summary>
        /// Gets or sets the treatment1 of the visit.
        /// </summary>
        public string Treatment1 { get; set; }

        /// <summary>
        /// Gets or sets the treatment2 of the visit.
        /// </summary>
        public string Treatment2 { get; set; }

        /// <summary>
        /// Gets or sets the treatment3 of the visit.
        /// </summary>
        public string Treatment3 { get; set; }

        /// <summary>
        /// Gets or sets the treatment4 of the visit.
        /// </summary>
        public string Treatment4 { get; set; }

        /// <summary>
        /// Gets or sets the treatment5 of the visit.
        /// </summary>
        public string Treatment5 { get; set; }

        /// <summary>
        /// Gets or sets the treatment6 of the visit.
        /// </summary>
        public string Treatment6 { get; set; }

        /// <summary>
        /// Gets or sets the Archive of the Visit.
        /// </summary>
        public bool Archive { get; set; }

        /// <summary>
        /// Gets or sets the PreventiveTreatmentsList of the Visit.
        /// </summary>
        public List<PreventiveTreatment> PreventiveTreatmentsList { get; set; }

        /// <summary>
        /// Keeps the database User of the animal owner.
        /// </summary>
        public ApplicationUserMysqlResponse User { get; set; }
    }
}
